lnwallet/chanfunding/wallet_assembler.go

```diff
@@ -299,11 +299,16 @@ func (w *WalletAssembler) ProvisionChannel(r *Request) (Intent, error) {
 			}
 		}
 
+		// Sanity check: The addition of the outputs should not lead to the
+		// creation of dust.
+		if changeAmt <= w.cfg.DustLimit {
```

Comment:
If this is a sanity check, should return an error instead?

Reply:
Ah yes, I have changed it to return an error instead.